Jumpscale/clients/blockchain/tfchain/TFChainReservation.py

```diff
@@ -0,0 +1,105 @@
+from Jumpscale import j
+import nacl
+
+_RESERVATION_SCHEMA = j.data.schema.get("""
+@url = tfchain.reservation
+type = "S3,VM" (E)
+duration = (I)
+size = (I)
+email = (email)
+bot_id = (S)
+created = (I)
+""")
+
+_GRID_BROKER_PUB_KEY = "..."
+
+
+class TFChainReservation():
+    """
+    TFChainReservation contains all TF grid reservation logic.
+    """
+
+    def __init__(self, wallet):
+        self._wallet = wallet
+        self._notary_client = None
+        self._grid_broker_pub_key = None
+        self._grid_broker_addr = ''  # TODO: hardcode tfchain address of the grid broker
+        # this all key management needs to be improved
+        keypair = self._wallet.key_pair_get(self._wallet.address)
+        self._signing_key = nacl.signing.SigningKey(keypair.private_key.to_seed())
+        self._private_key = self._signing_key.to_curve25519_private_key()
+
+    @property
+    def notary_client(self):
+        if self._notary_client is None:
+            self._notary_client = j.clients.gedis.configure('tfnotary', host='notary.grid.tf', port=8888)
+        return self._notary_client
+
+    @property
+    def grid_broker_pub_key(self):
+        if self._grid_broker_pub_key is None:
+            record = self._wallet.client.threebot.record_get('tf3bot.zaibon')
+            encoded_key = record.public_key.unlockhash
```

Comment:
I guess you know, but the unlock hash is more than just a hex encoding of the public key (hence why it is 78 bytes instead of 64 bytes).

Reply:
no I didn't know, can you provide a snippet how I should extract the pubic key of a 3bot please.

Reply:
`record.public_key.hash` returns the internal hash object, `str(record.public_key.hash)` returns these bytes hex encoded (64 bytes), and `record.public_key.hash.value` returns the actual bytes (32).